### PR TITLE
Enable CI

### PR DESCRIPTION
**Pull request by AbbyGi**

### CODE REVIEW

**Review by danielballan — APPROVED** [review]

File: .github/workflows/black.yml [hunk 1]
```diff
@@ -0,0 +1,11 @@
...
+
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - uses: actions/setup-python@v2
+      - uses: psf/black@stable
```
Comment (mrakitin) [inline .github/workflows/black.yml:11]: `black` action was broken some time ago, and was causing CI failures. In https://github.com/bluesky/bluesky-queueserver/blob/main/.github/workflows/black.yml we switched to a direct installation. Not sure what's best, but probably it's better to rely on the static (and ~always available) PyPI releases, than on a fragile branch on GH. Thoughts are welcome.

File: .github/workflows/testing.yml [hunk 2]
```diff
@@ -0,0 +1,42 @@
+name: Unit Tests
+
+on:
+  push:
+  pull_request:
+  schedule:
+      - cron: '00 4 * * *'  # daily at 4AM
```
Comment (mrakitin) [inline .github/workflows/testing.yml:7]: Do we need that frequent testing? Maybe weekly would make sense.

File: .github/workflows/testing.yml [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+
+jobs:
+  build:
+
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        python-version: [3.6, 3.7, 3.8]
```
Comment (mrakitin) [inline .github/workflows/testing.yml]: Do we still care about Python 3.6? I saw some of our projects got rid of it already, and rather use Python 3.7+, including 3.9.

File: .github/workflows/testing.yml [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+    - uses: actions/checkout@v2
+
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+
+    - name: Install
+      shell: bash -l {0}
+      run: source continuous_integration/scripts/install.sh
```
Comment (mrakitin) [inline .github/workflows/testing.yml:28]: I personally prefer short and hidden names for CI dirs, i.e. `.ci`, but I won't object too much here, if that approach is used in other places.

I think the advantage of the hidden dirs is that it does not mix with the code dirs when listed. But it's up to you.

File: .github/workflows/testing.yml [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+      run: |
+        set -vxeuo pipefail
+        python -m pip install -r requirements-dev.txt
+        python -m pip list
+
+    - name: Test with pytest
+      shell: bash -l {0}
+      run: |
+        set -vxeuo pipefail
+        coverage run -m pytest -v
```
Comment (mrakitin) [inline .github/workflows/testing.yml:41]: Do we still care about `coverage`? I thought it had a security issue, and people started abandoning it...

File: continuous_integration/scripts/install.sh [hunk 6]
```diff
@@ -0,0 +1,13 @@
...
+set -vxeuo pipefail
+
+# These packages are installed in the base environment but may be older
+# versions. Explicitly upgrade them because they often create
+# installation problems if out of date.
+python -m pip install --upgrade pip setuptools wheel numpy
+# Versioneer uses the most recent git tag to generate __version__, which appears
+# in the published documentation.
+git fetch --tags
+python -m pip install .
```
Comment (mrakitin) [inline continuous_integration/scripts/install.sh]: Optional suggestion for a more verbose installation the same way as it's done in our conda feedstocks. Feel free to ignore.
```suggestion
python -m pip install . -v
```

**Review by mrakitin — COMMENTED** [review]
Looks good! Thanks for the hard work of getting the modern CI into that project. I have a few suggestions based on considerations for https://github.com/bluesky/bluesky-queueserver/tree/main/.github/workflows.

File: .github/workflows/testing.yml (hunk 2 above)
Comment (mrakitin) [reply .github/workflows/testing.yml:7]: A side note: I really like that website to validate my crontab entries: [`https://crontab.guru/#00_4_*_*_*`](https://crontab.guru/#00_4_*_*_*).

Comment (danielballan) [reply .github/workflows/testing.yml:7]: We do daily currently to quickly pick up problems like breakages upstream. I could be worth re-evaluating but let's do that consistently across all projects.

File: .github/workflows/testing.yml (hunk 4 above)
Comment (danielballan) [reply .github/workflows/testing.yml:28]: I kind of agree with your point, but this is a pattern that we have established.

File: .github/workflows/testing.yml (hunk 5 above)
Comment (danielballan) [reply .github/workflows/testing.yml:41]: We weren't exposed to the security issue (which was specific to the bash uploader).

It's reasonable to discuss whether we want to continue to use coverage, but we currently do so in most (all?) repos --- see [bluesky/bluesky](https://github.com/bluesky/bluesky/blob/ebffcc55c7d28e279ce101cb79ea1ff4a3d92411/.github/workflows/testing.yml#L59-L60) --- so I think we should re-evaluate in a separate conversation and stay consistent for now.

File: .github/workflows/black.yml (hunk 1 above)
Comment (danielballan) [reply .github/workflows/black.yml:11]: Either way is fine with me.

Comment (danielballan) [reply .github/workflows/black.yml:11]: I agree PyPI is slightly better.

File: .github/workflows/testing.yml (hunk 3 above)
Comment (prjemian) [reply .github/workflows/testing.yml]: Is Py3.6 EOL yet?

**Pull request merged**